src/main/java/com/cleanroommc/groovyscript/compat/mods/tinkersconstruct/Casting.java

```diff
@@ -110,35 +116,44 @@ public boolean removeByCast(IIngredient cast) {
             return false;
         }
 
+        @MethodDescription(description = "groovyscript.wiki.removeAll")
         public void removeAll() {
             TinkerRegistryAccessor.getTableCastRegistry().forEach(this::addBackup);
             TinkerRegistryAccessor.getTableCastRegistry().forEach(TinkerRegistryAccessor.getTableCastRegistry()::remove);
         }
 
+        @MethodDescription(description = "groovyscript.wiki.streamRecipes")
         public SimpleObjectStream<ICastingRecipe> streamRecipes() {
             return new SimpleObjectStream<>(TinkerRegistryAccessor.getTableCastRegistry()).setRemover(this::remove);
         }
 
         public class RecipeBuilder extends AbstractRecipeBuilder<ICastingRecipe> {
 
+            @Property
             private IIngredient cast;
+            @Property(defaultValue = "200", valid = @Comp(value = "1", type = Comp.Type.GTE))
             private int time = 200;
+            @Property(defaultValue = "false")
             private boolean consumesCast = false;
 
+            @RecipeBuilderMethodDescription(field = "time")
             public RecipeBuilder coolingTime(int time) {
                 this.time = Math.max(time, 1);
                 return this;
             }
 
+            @RecipeBuilderMethodDescription(field = "consumesCast")
```

Comment:
if the `field` value isnt set, it will default to the method name, meaning any method where the method is identical to the field modified is optional